pkg/arvo/app/dojo.hoon

```diff
@@ -1348,8 +1354,14 @@
     ::  ~&  [%his-clock ler.cal]
     ::  ~&  [%our-clock ven.say]
     =^  dat  say  (~(transceive sole say) cal)
-    ?.  ?&  ?=(%del -.dat)
-            =(+(p.dat) (lent buf.say))
+    =/  len  (lent buf.say)
+    ?.  ?|  ?&  ?=(%del -.dat)
+                =(+(p.dat) len)
+            ==
+            ?&  ?=([%mor ^] dat)
+                =(%del +<-:dat)
+                =(+(;;(@ud +<+:dat)) len)
+            ==
```

Comment:
```suggestion
            ?&  ?=([%mor [%del *] *] dat)
                =(+(p.i.p.dat) len)
            ==
```

You can type-check for the exact thing you want in one go, and then access the `%del` value using faces instead of lark.